### ISSUE TITLE
Rule proposal: `prefer-array-some` 

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#736]
# Fail

```js
if (array.find(element => element === '🦄')) {
}
```

```js
const foo = array.find(element => element === '🦄') ? bar : baz;
```

# Pass

```js
if (array.some(element => element === '🦄')) {
}
```

```js
const foo = array.some(element => element === '🦄') ? bar : baz;
```

Just experienced one  few days ago, and kind of follow up for https://github.com/sindresorhus/eslint-plugin-unicorn/issues/730

**Comment by fisker** [comment]
One note, we  should definitely filter the first argument which is `Literal` like https://github.com/sindresorhus/eslint-plugin-unicorn/pull/704#issuecomment-623667663 and  https://github.com/sindresorhus/eslint-plugin-unicorn/pull/697, because there is `jQuery.find()`.

**Comment by silverwind** [comment]
I wouldn't specifically suggest `some` but warn on the unused `array#find` return value. I did a rule to eslint proposal regarding a general unused return values rule a while ago, maybe it is worth exploring here:

https://github.com/eslint/eslint/issues/12980

**Comment by papb** [comment]
> I wouldn't specifically suggest `some` but warn on the wrongly used `array#find` return value in the boolean context. 

I like the idea of suggesting `some`. Might help users who don't know any better.

> BTW, I did a rule to eslint proposal regarding a general unused return values rule a while ago, maybe it is worth exploring here, #741.

@silverwind What is the connection? In this case, the return value is being used, so what exactly do you mean?

**Comment by silverwind** [comment]
There's no real connection, my mistake. Return value is used here (thought wrongly).

**Comment by silverwind** [comment]
My issue with `array#some` and `array#every` is that I think it's hard to read/understand. I prefer `for-of` with a `break` statement because I don't have to think about whether the iterator needs to be `true` or `false` to break the loop.

**Comment by fisker** [comment]
Why are you thinking about breaking loop? When use `some` and `every` you should think  is this element/result I'm searching? One important thing you should think about is, if it's empty, what the result will be.

**Comment by silverwind** [comment]
Yeah it might just be me and `some`/`every` are fine. As an example, I'd prefer the second variant here because it just seems easier to read to me:

```js
function hasBiggerThan10(array) {
  return array.some(element => element > 10);
}

function hasBiggerThan10(array) {
  for (const element of array) {
    if (element > 10) return true;
  }
  return false;
}

**Comment by fisker** [comment]
```js

const isBiggerThan10 = element => element > 10;

function hasBiggerThan10(array) {
  return array.some(isBiggerThan10);
}

function allBiggerThan10(array) {
  return array.every(isBiggerThan10);
}

```

**Comment by silverwind** [comment]
Maybe I've expressed myself badly but the "hard" part is remembering to return a truthy value in the callback.

One distinct advantage `for-of` has over array methods is that it works on all iterables, so it's a win-win to use it over array methods.

**Comment by fisker** [comment]
Yes, I  agree in some case loop  better,  some also  useful to avoid use `let`/`break` in  simple tests. I  use both.

**Comment by riophae** [comment]
Although this could be a taste issue, `Array#some` can easily run on all iterables too. Just FYI.

```js
Array.from(set).some(...)
```

**Comment by sindresorhus** [comment]
This is accepted.

**Comment by fisker** [comment]
WIP
